distribution/docs/src/main/resources/_contents/_endpoints/cometd-endpoint-contents.adoc

```diff
@@ -0,0 +1,855 @@
+* subscribing to `/${branding-lowercase}/notifications/catalog/downloads` will cause the client to only receive notifications of downloads.
+
+====== Using CometD Notifications
+
+[NOTE]
+====
+The ${branding} Search UI serves as a reference implementation of how clients can use notifications.
+====
+
+Notifications are currently being utilized in the Catalog application for resource retrieval.
+When a user initiates a resource retrieval, the channel `/ddf/notification/catalog/downloads` is opened, where notifications indicating the progress of that resource download are sent.
+Any client interested in receiving these progress notifications must subscribe to that channel.
+
+${branding} starts downloading the resource to the client that requested it, a notification with a status of "Started" will be broadcast.
+If the resource download fails, a notification with a status of "Failed" will be broadcast.
+Or, if the resource download is being attempted again after a failure, "Retry" will be broadcast.
+When a notification is received, ${branding} Search UI displays a popup containing the contents of the notification, so a user is made aware of how their downloads are proceeding.
+Behind the scenes, the ${branding} Search UI invokes the REST endpoint to retrieve a resource.
+
+In this request, it adds the query parameter "user" with the CometD session ID or the unique User ID as the value.
+This allows the CometD server to know which subscriber is interested in the notification.
+For example, `\http://${branding}_HOST:8181/services/catalog/sources/${branding-lowercase}.distribution/2f5db9e5131444279a1293c541c106cd?
+  transform=resource&user=1w1qlo79j6tscii19jszwp9s2i55` notifications contain the following information:
+
+.Notification Contents
+[cols="1m,3,1" options="header"]
+|===
+|Parameter Name
+|Description
+|Always Included with Notification
+
+|application
+|"Downloads" for resource retrieval. This is used as a "type" or category of messages.
+|Yes
+
+|title
+|Resource/file name for resource retrieval.
+|Yes
+
+|message
+|Human-readable message containing status details.
+|Yes
+
+|timestamp
+|Timestamp in milliseconds of when event occured.
+|Yes
+
+|user
+|CometD Session ID or unique User ID.
+|Yes
+
+|status
+|Status of event.
+|No
+
+|option
+|Resource retrieval option.
+|No
+
+|bytes
+|Number of bytes transmitted.
+|No
+
+|===
+
+====== Receive Notifications
+
+* If interested in retrieve resource notifications, a client must subscribe to the CometD channel `/ddf/notification/catalog/downloads`.
+* If interested in all notification types, a client must subscribe to the CometD channel `/ddf/notification/\*\*`
+* A client will only receive notifications for resources they have requested.
+* Standard UI is subscribed to all notifications of interest to that user/browser session: `/ddf/notification/\*\*`
+* See the Usage section for the data that a notification contains.
```

Comment:
I'll put an anchor on the table. The table numbers are assigned dynamically and can change if something is added/removed upstream of this section.